src/walletSdk/Anchor/index.ts

```diff
@@ -201,6 +224,183 @@ export class Anchor {
     }
   }
 
+  /**
+  * Watch all transactions returned from a transfer server. When new or
+  * updated transactions come in, run an `onMessage` callback.
+  *
+  * On initial load, it'll return ALL pending transactions via onMessage.
+  * Subsequent messages will be any one of these events:
+  *  * Any new transaction appears
+  *  * Any of the initial pending transactions change any state
+  *
+  * You may also provide an array of transaction ids, `watchlist`, and this
+  * watcher will always react to transactions whose ids are in the watchlist.
+  */
+  watchAllTransactions(params: {
+    authToken: string;
+    assetCode: string;
+    onMessage: (transaction) => void;
+    onError: (error: any) => void;
+    watchlist?: string[];
```

Comment:
So, when transaction is in the watchlist, we always call onMessage with this transaction, even if status didn't change, right?
What's the use-case for that?

Reply:
Kind of, we always call `onMessage` on the **first try** for watchlist transactions, but on subsequent retries we only call `onMessage` again [if there is a change in the transaction](https://github.com/stellar/typescript-wallet-sdk/pull/12/files#diff-08c9260268921699d5ac092cdf53654f59685a9378df00a964422ec206c4c419R318) otherwise it does nothing (i.e. keep silently waiting until there is a transaction change).

Humm I don't have any specific/strong use-case for it so far, but thought it could be useful for wallets that wish to watch for several transactions on an id-basis instead of a status-basis.

That being said, I'm not against removing it in case we'd like to keep it simple for now (and add it later on the road in case we get some clear use cases for it).

Reply:
Got it. I don't have a strong preference, it's up to you if you want to remove it or keep it.
Agree that keeping it simple for now is a good idea, but leaving it be is also OK with me